app/src/test/java/io/intrepid/contest/screens/contestcreation/editcategoriestocontest/EditCategoriesPresenterTest.java

```diff
@@ -33,11 +36,30 @@ public void presenterCanBeCreated() {
     }
 
     @Test
-    public void onNextClickedShouldTriggerViewToAddCategory() {
+    public void onViewCreatedShouldDisplayPreExistingCategoryDetails() {
+        editCategoryPresenter.onViewCreated();
+        verify(mockView).showCategory(any(), any());
+    }
+
+    @Test
+    public void onNextClickedShouldTriggerViewToEditCategory() {
+        String categoryName = "NewCategory";
+        String categoryDescription = "NewCategory Description";
+        editCategoryPresenter.onNextClicked(categoryName, categoryDescription);
+        verify(mockView).editCategory(any(), eq(categoryName), eq(categoryDescription));
+    }
+
+    @Test
+    public void onNextClickedShouldTriggerViewToAddCategoryWhenParamsAreNull() {
+        editCategoryPresenter = new EditCategoriesPresenter(mockView,
+                                                            TestPresenterConfiguration.createTestConfiguration(),
```

Comment:
Would using `testConfiguration` conflict with previously assigned presenter?

Same as before, you can comment you're testing non-editMode here.